lib/segment/src/types.rs

```diff
@@ -293,6 +293,37 @@ impl From<PayloadInterfaceStrict> for PayloadInterface {
     }
 }
 
+impl From<&PayloadType> for PayloadInterface {
+    fn from(payload: &PayloadType) -> Self {
+        match payload {
+            PayloadType::Keyword(x) => {
+                match x.len() {
+                    1 => PayloadInterface::KeywordShortcut(PayloadVariant::Value(x.first().unwrap().to_string())),
+                    _ => PayloadInterface::KeywordShortcut(PayloadVariant::List(x.clone())),
+                }
+            }
+            PayloadType::Integer(x) => {
+                match x.len() {
+                    1 => PayloadInterface::IntShortcut(PayloadVariant::Value(*x.first().unwrap())),
+                    _ => PayloadInterface::IntShortcut(PayloadVariant::List(x.clone())),
+                }
+            }
+            PayloadType::Float(x) => {
+                match x.len() {
+                    1 => PayloadInterface::FloatShortcut(PayloadVariant::Value(*x.first().unwrap())),
+                    _ => PayloadInterface::FloatShortcut(PayloadVariant::List(x.clone())),
+                }
+            }
+            PayloadType::Geo(x) => {
+                match x.len() {
+                    1 => PayloadInterface::Payload(PayloadInterfaceStrict::Geo(PayloadVariant::Value(x.first().unwrap().clone()))),
+                    _ => PayloadInterface::Payload(PayloadInterfaceStrict::Geo(PayloadVariant::List(x.clone())))
+                }
+            }
+        }
+    }
+}
+
```

Comment:
That could create an inconsistency: If payload was created like `{"field": ["value"]}` it will be returned as `{"field": "value"}`. And I would prefer to have a consistent schema over a good-looking result.